woocommerce/woocommerce
# Template API: Expression evaluation package

**Yes, this PR is large. But, it is mostly lock file updates, documentation, and unit tests!**

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR implements support for evaluating JavaScript-like expressions in an optional context, using [Peggy](https://peggyjs.org/index.html) to construct a grammar and generate a parser from it.

These expressions will be used to support conditional visibility in the template API in a follow up PR (see issues #40232).

Closes #40234.



### How to test the changes in this Pull Request:



1. Verify unit tests pass (`pnpm run test --filter=@woocommerce/expression-evaluation`)
2. Verify `README.md` contains accurate documentation (the unit tests should cover all cases, but you can write your own additional tests to confirm things if need be)

**The package introduced in this PR is not used by any other package yet, so there are no changes to WooCommerce itself.**



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
